Clarify that attribute keys are unique in collections (#2248)
open-telemetry/opentelemetry-specification: specification/common/common.md

Attributes, which belong to Metrics, are exempt from the limits described above
at this time, as discussed in
[Metrics Attribute Limits](../metrics/sdk.md#attribute-limits).

## Attribute Collections

> [Resources](../resource/sdk.md), Metrics
> [data points](../metrics/datamodel.md#metric-points),
> [Spans](../trace/api.md#set-attributes), Span
> [Events](../trace/api.md#add-events), Span
> [Links](../trace/api.md#specifying-links) and
> [Log Records](../logs/data-model.md) may contain a collection of attributes. The

Reviewer: ```suggestion
[Resources](../resource/sdk.md), 
Metrics [data points](../metrics/datamodel.md#metric-points),
[Spans](../trace/api.md#set-attributes), 
Span [Events](../trace/api.md#add-events), 
Span [Links](../trace/api.md#specifying-links) 
and [Log Records](../logs/data-model.md)
may contain a collection of attributes. The
```
this formatting makes this more readable

Author: Done.

> The keys in each such collection are unique, i.e. there cannot exist more than

Reviewer 2: cannot
```suggestion
The keys in each such collection are unique, i.e. there MUST NOT exist more than
```

And we probably should specify that conflicts SHOULD be resolved by using the attribute set latest.

Reviewer 3: Yes, I believe this was widely discussed and agreed to this point _years ago_, we just never wrote it down clearly. When an attribute map is expressed as a list, duplicate keys should be resolved by taking the last value among the ordered list. This improves efficiency since you can let the receiver de-duplicate.

> one key-value pair with the same key.

Reviewer 4: Probably adding a link here to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-attributes would keep clear what part is responsible for enforcing it.

Author: Yes, line 114 links exactly to that.

> one key-value pair with the same key.

Reviewer 2: cannot
```suggestion
The keys in each such collection are unique, i.e. there MUST NOT exist more than
```

And we probably should specify that conflicts SHOULD be resolved by using the attribute set latest.
